src/wifi_hal_nl80211.c

```diff
@@ -11859,7 +11875,11 @@ int wifi_drv_sta_deauth(void *priv, const u8 *own_addr, const u8 *addr, u16 reas
           HOSTAPD_MODE_IEEE80211AD) {
         /* Deauthentication is not used in DMG/IEEE 802.11ad;
            * disassociate the STA instead. */
+#if defined(BANANA_PI_PORT) && defined(KERNEL_6_6)
+        return wifi_drv_sta_disassoc(priv, own_addr, addr, reason, -1);
+#else
```

Comment:
Please tell me why this is needed; 
On kernel 6.6 we want to have host ap >= 2.11 AND generic mlo support. 
Thus, link_id is usable (line 11753 / 11769) . If we are on kernel 5.5,  link_id will be set to -1; What is the reason for passing -1 and making separate definition of wifi_drv_sta_disassoc. 

If i understood it wrongly, then it looks to me that one extra ifdef will suffice, below this one:
#if HOSTAPD_VERSION >= 211 && defined(CONFIG_GENERIC_MLO)
    link_id = wifi_hal_get_mld_link_id(interface);
#endif // HOSTAPD_VERSION >= 211 && CONFIG_GENERIC_MLO
setting link_id to -1.  ?

Reply:
```suggestion
#if HOSTAPD_VERSION >= 211 && defined(CONFIG_GENERIC_MLO)
    link_id = wifi_hal_get_mld_link_id(interface);
#endif // HOSTAPD_VERSION >= 211 && CONFIG_GENERIC_MLO

#if defined(BANANA_PI_PORT) && defined(KERNEL_6_6)
    link_id = -1;
#endif
```
This is my suggestion - conditionally overwrite value, only on BPI+kernel 6_6. (after line 11769, not here). Then rest of the defines for wifi_drv_sta_disassoc() should be not needed.

or:
```suggestion
#if HOSTAPD_VERSION >= 211 && defined(CONFIG_GENERIC_MLO)
#if defined(BANANA_PI_PORT) && defined(KERNEL_6_6)
    link_id = -1;
#else
    link_id = wifi_hal_get_mld_link_id(interface);
#endif //BANANA_PI_PORT && KERNEL_6_6
#endif // HOSTAPD_VERSION >= 211 && CONFIG_GENERIC_MLO
```
Plus a line of comment, why this is needed (Since it's not obvious) and (if known) when can it be removed.